src/python/pants/backend/python/packaging/pyoxidizer/rules.py

```diff
@@ -0,0 +1,151 @@
+    PyOxidizerConfigSourceField,
+    PyOxidizerDependenciesField,
+    PyOxidizerEntryPointField,
+    PyOxidizerUnclassifiedResources,
+)
+from pants.backend.python.util_rules.pex import Pex, PexProcess, PexRequest
+from pants.core.goals.package import BuiltPackage, BuiltPackageArtifact, PackageFieldSet
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestContents,
+    FileContent,
+    MergeDigests,
+    Snapshot,
+)
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import (
+    DependenciesRequest,
+    FieldSetsPerTarget,
+    FieldSetsPerTargetRequest,
+    Targets,
+)
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PyOxidizerFieldSet(PackageFieldSet):
+    required_fields = (PyOxidizerDependenciesField,)
+
+    entry_point: PyOxidizerEntryPointField
+    dependencies: PyOxidizerDependenciesField
+    unclassified_resources: PyOxidizerUnclassifiedResources
+    template: PyOxidizerConfigSourceField
+
+
+@rule(level=LogLevel.DEBUG)
+async def package_pyoxidizer_binary(
+    pyoxidizer: PyOxidizer, field_set: PyOxidizerFieldSet
+) -> BuiltPackage:
+    logger.info(f"Incoming package_pyoxidizer_binary field set: {field_set}")
+    targets = await Get(Targets, DependenciesRequest(field_set.dependencies))
+    target = targets[0]
+
+    logger.info(
+        f"Received these targets inside pyox targets: {target.address.target_name}"
+    )
+
+    packages = await Get(
+        FieldSetsPerTarget,
+        FieldSetsPerTargetRequest(PackageFieldSet, [target]),
+    )
+    logger.info(f"Retrieved the following FieldSetsPerTarget {packages}")
+
+    built_packages = await MultiGet(
+        Get(BuiltPackage, PackageFieldSet, field_set)
+        for field_set in packages.field_sets
+    )
+
+    # TODO: Can this be walrus'd? Double for with repeated artifact.relpath is ugly
+    wheels = [
+        artifact.relpath
+        for wheel in built_packages
+        for artifact in wheel.artifacts
+        if artifact.relpath is not None
+    ]
+    logger.info(f"This is the built package retrieved {built_packages}")
+
+    # Pulling this merged digests idea from the Docker plugin
+    built_package_digests = [built_package.digest for built_package in built_packages]
+
+    # Pip install pyoxidizer
+    pyoxidizer_pex_get = await Get(
+        Pex,
+        PexRequest(
+            output_filename="pyoxidizer.pex",
+            internal_only=True,
+            requirements=pyoxidizer.pex_requirements(),
+            interpreter_constraints=pyoxidizer.interpreter_constraints,
+            main=pyoxidizer.main,
+        ),
+    )
+
+    config_template = None
+    if field_set.template.value is not None:
```

Comment:
For `isort` and `black`, the files are expected to be located at certain places in the repository, and so the config scraping in those cases is a convenience to avoid needing to specify a config. In this case, requiring a per-binary config seems completely reasonable, since I don't think that any two binaries are likely to share a config.